hole-punch-interop/rust-relay/Dockerfile

```diff
@@ -1,18 +1,33 @@
 # syntax=docker/dockerfile:1.5-labs
-FROM rust:1.72.0 as builder
 
-RUN rustup target add x86_64-unknown-linux-musl
-RUN --mount=type=cache,target=/var/cache/apt apt-get update && apt-get install -y musl-dev musl-tools
+FROM --platform=$BUILDPLATFORM rust:1.72.0 as builder
+
+ARG BUILDPLATFORM
+
+# Copy the setup.sh script into the image
+COPY setRustTarget.sh /setRustTarget.sh
+
+RUN chmod +x /setRustTarget.sh && /setRustTarget.sh ${BUILDPLATFORM}
+
+# Conditionally install musl-tools if targeting musl
+RUN --mount=type=cache,target=/var/cache/apt \
+    . /etc/environment && \
+    apt-get update && \
+    if [[ "${RUST_TARGET}" == *"-musl"* ]]; then \
+        apt-get install -y musl-dev musl-tools; \
```

Comment:
I'd just always install this, doesn't hurt and simplifies the dockerfile.

Reply:
Will use `aarch64-unknown-linux-musl` and do that.